runtime/kubernetes/kubernetes.go

```diff
@@ -86,19 +85,24 @@ func (k *kubernetes) Create(s *runtime.Service, opts ...runtime.CreateOption) er
 		o(&options)
 	}
 
-	svcName := s.Name
+	// quickly prevalidate the name and version
+	name := s.Name
 	if len(s.Version) > 0 {
-		svcName = strings.Join([]string{s.Name, s.Version}, "-")
+		name = name + "-" + s.Version
 	}
 
-	if !client.ServiceRegexp.MatchString(svcName) {
-		return fmt.Errorf("invalid service name: %s", svcName)
+	// format as we'll format in the deployment
+	name = client.Format(name)
+
+	// check the name to ensure what we pass into metadata will be validated
+	if !client.ServiceRegexp.MatchString(name) {
```

Comment:
I really do wonder whether we should be doing the `regexp` validation ourselves or whether we should just rely on K8s and return error if K8s validation fails 🤷‍♂️I feel there is a rabbit hole looming over here, but I might be wrong.

Reply:
Yea I agree. maybe we should just try to shovel cruft at k8s and expect that it does the right thing. Will strip it out.